homeassistant/helpers/storage.py

```diff
@@ -24,6 +24,8 @@
 
 STORAGE_SEMAPHORE = "storage_semaphore"
 
+_T = TypeVar("_T", bound=MutableMapping[str, Any])
```

Comment:
I am having trouble binding this correctly. I think we want to be able to accept any `list`, or a dict with `str` keys.
but if I use `TypeVar("_T", list, dict)` then it only accept `list[Any]` and `dict[Any, Any]`. It won't accept any other variant like `dict[str, str]` `dict[str, Any]`